src/Domains/Registrar/OpenSRS.php

```diff
@@ -0,0 +1,659 @@
+    ];
+
+    $result = $this->send($message);
+    $result = simplexml_load_string($result);
+    
+    $elements = $result->xpath('//body/data_block/dt_assoc/item[@key="is_success"]');
+    $successful = "{$elements[0]}" == '1' ? true : false;
+
+    return $successful;
+  }
+  
+  public function suggest(array $query, array $tlds = array(), $minLength = 1, $maxLength = 100)
+  {
+    $message = [
+      'object' => 'DOMAIN',
+      'action' => 'name_suggest',
+      'attributes' => [
+        'services' => [
+          'suggestion',
+        ],
+        'searchstring' => implode(' ', $query),
+        'tlds' => $tlds,
+      ]
+    ];
+
+    $xpath = implode('/', [
+      '//body',
+      'data_block',
+      'dt_assoc',
+      'item[@key="attributes"]',
+      'dt_assoc',
+      'item[@key="suggestion"]',
+      'dt_assoc',
+      'item[@key="items"]',
+      'dt_array',
+      'item',
+    ]);
+
+    $result = $this->send($message);
+    $result = simplexml_load_string($result);
+    $elements = $result->xpath($xpath);
+    
+    $items = [];
+    
+    foreach ($elements as $element) {
+      $item = $element->xpath('dt_assoc/item');
+      $domain = "{$item[0]}";
+      $available = "{$item[1]}" == 'available' ? true : false;
+
+      $items[$domain] = $available;
+    }
+
+    return $items;
+  }
+  
+  public function tlds():array
+  {
+    // OpenSRS offers no endpoint for this
+    return [];
+  }
+  
+  public function domain(string $domain)
+  {
+    $message = [
+      'object' => 'domain',
+      'action' => 'get',
+      'attributes' => [
+        'domain' => $domain,
+        'type' => 'all_info',
+        'clean_ca_subset' => 1
+      ]
+    ];
+
+
+    $xpath = implode('/', [
+      '//body',
+      'data_block',
+      'dt_assoc',
+      'item[@key="attributes"]',
+      'dt_assoc',
+      'item'
+    ]);
+
+    $result = $this->send($message);
+    $result = simplexml_load_string($result);
+    $elements = $result->xpath($xpath);
+
+    $results = [];
+
+    foreach($elements as $element) {
+      $key = "{$element['key']}";
+      $value = "{$element}";
+
+      $results[$key] = $value;
+    }
+
+    return $results;
+  }
+  
+  public function updateDomain(string $domain, array $details)
+  {
+    $message = [
+      'object' => 'domain',
+      'action' => 'modify',
+      'attributes' => [
+        'domain' => $domain,
+        'affect_domains' => 0,
+        'data' => $details['data'],
+        'contact_set' => $details['contact_set']
+      ]
+    ];
+
+
+    $xpath = implode('/', [
+      '//body',
+      'data_block',
+      'dt_assoc',
+      'item[@key="attributes"]',
+      'dt_assoc',
+      'item[@key="details"]',
+      'dt_assoc',
+      'item[@key="'.$domain.'"]',
+      'dt_assoc',
+      'item[@key="is_success"]'
+    ]);
+
+    $result = $this->send($message);
+    $result = simplexml_load_string($result);
+    $elements = $result->xpath($xpath);
+
+    return "{$elements[0]}" == '1' ? true : false;
```

Comment:
```suggestion
    return $elements[0] === '1';
```

should be enough right?

Reply:
If i don't put it in quotes, it doesn't give the same put put, I assume there is a `__toString` attached to it, other wise === just gives me the node/object itself.